jena-tdb/src/main/java/org/apache/jena/tdb/solver/BindingTDB.java

```diff
@@ -120,6 +121,8 @@ public Node get1(Var var)
             if ( id == null )
                 return null ; 
             n = nodeTable.getNodeForNodeId(id) ;
+            if ( n == null )
+                throw new TDBException("No node in NodeTable for NodeId "+id);
```

Comment:
Do we know all/most of the circumstances that might cause this? Is it basically an interrupted write? If so, it might be worth including some error logging with that hint, but if there are more than one or two potential causes, never mind. I mention it because getting from "a missing entry in the `NodeTable`" to "that was the aftereffect of a failed write" is going to bring some users to the list when they might be able to figure out what happened with a simple hint.

Reply:
I don't know all the reasons for getting it.  The use  reports don't tend to tell us when things are fixed so I don't for sure whether my suggestions were of any help.

One possible case is early exit, non-transactional use of TDB so the triple or quad table has written back (by natural OS managed file caching) but the NodeTable has not.  Exit without sync (not necessarily a crash) and they are out of step.

Maybe with this we'll get better information and be able to refine it - at the moment, coudl well be a false trail.

Reply:
Right-o, sounds like it's much better to leave it be.